docs/docs/security-in-gatsby.md

```diff
@@ -1,9 +1,95 @@
+  // highlight-next-line
+  <p dangerouslySetInnerHTML={{ __html: comment }} />
+) // dangerous indeed.
+```
+
+**How can it be prevented?**
+
+The most straightforward way to prevent the XSS attack is to sanitize the innerHTML string before dangerously setting it. Fortunately, there are npm packages that does the job, like [sanitize-html](https://www.npmjs.com/package/sanitize-html) and [DOMPurify](https://github.com/cure53/DOMPurify).
+
+## Cross-Site Request Forgery (CSRF)
+
+A web application that use cookies could be attacked by the CSRF exploit, which deceives a your user to execute actions by its name. By default, the browser "trusts" all the activity made validating the user's identity and therefore sending the associated cookies in every request.
+
+For example, assume that your users received an email with a malicious link:
+
+```html
+<a href="http://maliciouswebsite.xyz/">Get your coupon here</a>
+```
+
+Then, when the website opens, it could render this tag:
+
+```html
+<img src="http://yourecommerce.com/profile" />
+```
+
+which, instead of rendering an image, it sends the request with the authentication cookies and opens a security breach in their personal information.
+
+**How can it be prevented?**
+
+This could be prevented in three major ways:
+
+### Additional Session Token
+
+In addition to cookie validation, another approach could be added in the flow, such as an [OAuth2](https://oauth.net/2/).
+
+### Same-Site Cookies
+
+Configure the created cookie in a way that it could be only sent by the same domain. [Most moderns browsers support it with a backward compatibility](https://caniuse.com/#feat=same-site-cookie-attribute).
+
+### Anti-CSRF Tokens
+
+In addition to a form, a hidden input could be added, holding the value of a encrypted token generated by the serve when page loads. When the form is sent, the server can block the action if they are not the same.
+
+## Third-party Scripts
+
+Some third-party scripts like Google Tag Manager gives the possibility to add arbitrary JavaScript code on top of them. Be sure to control the access of these services.
+
+## Check Your Dependencies
+
+In your Gatsby project, you are going to have a lot of dependencies (which has their own dependencies as well) in your `node_modules/`. Therefore, it is important to check if any of them has a security issue. NPM make this possible since v6 by executing the `npm audit` in the command line.
+
+## Content Security Policy (CSP)
+
+Content Security Policy is a security layer added in web applications to detect and prevent attacks, e.g. the XSS attack mentioned above.
+
+To add it to your Gatsby website, add the [plugin](https://www.gatsbyjs.org/packages/gatsby-plugin-csp/) to your `gatsby-config.js` with the desired configuration. Note that
+currently there is a [incompatibility issue](https://github.com/gatsbyjs/gatsby/issues/10890) with [gatsby-image](https://www.gatsbyjs.org/packages/gatsby-image) and other plugins.
```

Comment:
Expound on  "other plugins"...I understand it's unreasonable to list every incompatible plugin, but please define _why_ or _what_ makes a plugin incompatible.

Reply:
Sure thing! I wrote the reason after that, which is more detailed in the issue.